projects/plugins/jetpack/extensions/blocks/nextdoor/edit.js

```diff
@@ -93,23 +93,45 @@ export function NextdoorEdit( {
 		</Placeholder>
 	);
 
-	const blockPreview = () => {
-		return (
-			<>
-				<div className={ `${ defaultClassName }-overlay` }></div>
-				<iframe width="100%" height="200" frameBorder="0" src={ iframeUrl } title="Nextdoor" />
-			</>
-		);
-	};
-
 	return (
 		<div className={ className }>
 			<NextdoorControls
 				{ ...{ defaultClassName, nextdoorShareUrl, onFormSubmit, setNextdoorShareUrl } }
 			/>
-			{ iframeUrl ? blockPreview() : blockPlaceholder }
+			{ iframeUrl ? (
+				<BlockPreview defaultClassName={ defaultClassName } iframeUrl={ iframeUrl } />
+			) : (
+				blockPlaceholder
+			) }
 		</div>
 	);
 }
 
+const BlockPreview = ( { defaultClassName, iframeUrl } ) => {
+	const html = `
+		<script>
+			window.addEventListener( 'message', function ( event ) {
+				if ( ! event.origin.startsWith( 'https://nextdoor' ) ) {
+					return;
+				}
+				var iframe = document.getElementById( 'nextdoor-embed' );
+				if ( event.source !== iframe.contentWindow ) {
+					return;
+				}
+				iframe.setAttribute( 'height', event.data.height );
+				var clientBoundingRect = document.body.getBoundingClientRect();
+				window.parent.postMessage( { action: 'resize', width: clientBoundingRect.width, height: clientBoundingRect.height }, '*' );
+			} );
+		</script>
+		<iframe id="nextdoor-embed" width="100%" height="200" frameBorder="0" src="${ iframeUrl }" title="Nextdoor" />
+		`;
```

Comment:
This is a bit of a mess, but the resize events weren't firing properly in the sandbox so I've ended up having to mimic the the resize functionality in [the SandBox component](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/sandbox/index.tsx#L24-L35).

Overall I think using a SandBox element is the way to go, as then we're in control of the iframe context, and can receive the messages.